src/codegen/mod.rs

```diff
@@ -1450,7 +1449,7 @@ impl CodeGenerator for CompInfo {
             ctx.options().derive_copy
         {
             derives.push("Copy");
-            if used_template_params.is_some() {
+            if used_template_params.is_empty() {
```

Comment:
I guess we must have been hitting `Some(v)` where `v.is_empty()` fairly often.

Reply:
Should this be `!used_template_params.is_empty()`? When the ! is put in the `impl clone`'s are left alone when the tests are run.

Reply:
Totally. Should have caught this earlier! Good eye ;)